### DISCUSSION TITLE
Mesh will not load no matter the start gcode

### DISCUSSION THREAD

**Discussion by asoundperson** [Jyers/Marlin#434] [Q&A] [answered]
The mesh stored, which I am able to view from the menu, does not load.
I tried:
M420 L1 and S1, on their own and followed by G29 J
Tried G29 A followed by L1 and J.

In a different firmware (smith3d) I got it working with a g29 followed by m500 followed by m420 L1.

All of these all present different results: sometimes it looks like it is the correct z height but with no UBL applied (I know because the back of the bed is about 0.5mm higher than the front and with the skirt you can visually see that), sometimes it adds 3/4mm to the above, sometimes it will go way too close to the bed.

I observe the above with a large skirt.

What can I try? Am I missing something silly?

**Comment by xantrk** [comment]
Load command should have the index of "0", not "1". My start gcode from prusaslicer is:

G90 ; use absolute coordinates
M83 ; extruder relative mode
M140 S[first_layer_bed_temperature] ; set bed temp
M190 S[first_layer_bed_temperature] ; wait for bed temp
G28 ; home all
G29 A ; Activate UBL
G29 L0 ; Load the mesh stored
G29 J ; Probe 3 points to tilt mesh
G92 E0 ; Reset Extruder
G1 Z20.0 F3000 ; Move Z Axis up to 20mm to prevent scratching of Heat Bed
G1 X0.1 Y15 Z20.0 F5000.0 ; Move to start position at z2
G1 Z0.5 ;  get close to the bed
M104 S[first_layer_temperature] ; set extruder temp
M109 S[first_layer_temperature] ; wait for extruder temp
G1 X0.1 Y20 Z0.3 F5000.0 ; Move to start position
G1 X0.1 Y200.0 Z0.3 F1500.0 E15 ; Draw the first line
G1 X0.4 Y200.0 Z0.3 F5000.0 ; Move to side a little
G1 X0.4 Y20 Z0.3 F1500.0 E30 ; Draw the second line
G92 E0 ; Reset Extruder
G1 Z2.0 F3000 ; Move Z Axis up little to prevent scratching of Heat Bed
G1 X5 Y20 Z0.3 F5000.0 ; Move over to prevent blob squish

**Comment by asoundperson** [comment]
I got this to work on 1.3.4 with this bit of start GCode for the mesh part.

G29 A
G29 L0
G29 J2

I updated to 1.3.5.b, created a new mesh, and now after homing and tilting 4 points it starts to print 1cm up in the air.
Should I try to downgrade fw?

**Reply by kted** [reply]
Make sure you select Slot 0 to save the mesh.

**Reply by asoundperson** [reply]
![IMG_20210912_181529](https://user-images.githubusercontent.com/82235415/132997886-32f98b3e-81d3-4eff-a9df-a8288d96eb6c.jpg)

Yes it is saved in slot 0

**Comment by asoundperson** [comment, marked as answer]
Both M420 L0 and G29 L0 do not work. I also tried both on L1 just in case but I know that the mesh is saved on 0.

I have reflashed 1.3.5.b, still not working. I have deleted EEPROM and tried again, same. I noticed in 1.3.5.b if tried to change the slot from 0 to 1 for example I could not (in the picture above)

I have downgraded to 1.3.4, now it does now work anymore. Before I was on 1.3.4 on a working status, after upgrade then downgrade it does not. Only thing that changed is that I have re created a mesh from screatch and now using home offset from the printer rather than G92 but that shouldn't be related.

Wether start code has 
**G29 A
G29 L0
G29 J2**

or **just G29 A**

or **removing the tilting**

or **none of the above**

the behaviour of the printer is the same.

**Reply by ixendrax** [reply]
Is your z offset set? It looks vaguely 0.00 in your photo.

**Reply by asoundperson** [reply]
Aargh. It reset itself when I flashed and didnt set it back. I spiraled for the whole day and it was just there. Thank you. I feel so stupid.
